pkgs/development/compilers/rust/rustfmt.nix

```diff
@@ -0,0 +1,89 @@
+{ stdenv, rustPlatform, rustc, Security }:
+
+rustPlatform.buildRustPackage rec {
+  name = "rustfmt-${version}";
+  inherit (rustc) version src;
+
+  # the rust source tarball already has all the dependencies vendored, no need to fetch them again
+  cargoVendorDir = "vendor";
+  preBuild = "pushd src/tools/rustfmt";
+  postBuild = "popd";
```

Comment:
This causes the checkPhase to test rustc instead of rustfmt. This might be the causes of the errors you encountered.

Reply:
```suggestion
  preInstall = "popd";
```